### ISSUE TITLE
Error: Socket Hang up when electron-builder Release on github.

### ISSUE THREAD

**Issue by TusharKanjariya** [electron-userland/electron-builder#4940]
* **Version**: 22.6.0

* **Electron Version**:8.1.1
* **Electron Type (current, beta, nightly)**:Current

* **Target**: NSIS Windows

My `electron-builder.json` file for configurations

```javascript
{
  "appId": "com.screenshot.app",
  "copyright": "Copyright © 2019-Tushar Kanjariya",
  "productName": "Screenshot App",
  "nsis": {
    "perMachine": true,
    "oneClick": false,
    "allowToChangeInstallationDirectory": false,
    "license": "LICENSE.txt",
    "allowElevation": true
  },
  "publish": {
    "provider": "github",
    "owner": "TusharKanjariya",
    "repo": "screenshot-app-electron",
    "token": "my_token"
  }
}
```

When I run the command `npm run publish:github` then the following error will occur after the final build and uploading

![image](https://user-images.githubusercontent.com/40752660/81400691-57cf4e80-914b-11ea-892d-2d009c2f03fc.png)

Plz, Help to solve my Issue.

**Comment by b-zurg** [comment]
Is this consistently reproducable?  This seems like it could be related: https://github.com/request/request/issues/2047

Have you tried upgrading your node version?

**Comment by TusharKanjariya** [comment]
@b-zurg 
My Current Node version is v13.13.0.
I have tried this solution but can't able to solve the issue.

**Comment by b-zurg** [comment]
Question - how long does the upload take? This sort of error is something I've encountered elsewhere on long uploads

**Comment by hovancik** [comment]
I am having the same today, never had this issue before: https://travis-ci.com/github/hovancik/stretchly/jobs/362382915

**Comment by hovancik** [comment]
```
⨯ Cannot cleanup: 

Error #1 --------------------------------------------------------------------------------

Error: socket hang up

    at connResetException (internal/errors.js:570:14)

    at TLSSocket.socketOnEnd (_http_client.js:440:23)

    at TLSSocket.emit (events.js:228:7)

    at endReadableNT (_stream_readable.js:1185:12)

    at processTicksAndRejections (internal/process/task_queues.js:81:21)

Error #2 --------------------------------------------------------------------------------

Error: socket hang up

    at connResetException (internal/errors.js:570:14)

    at TLSSocket.socketOnEnd (_http_client.js:440:23)

    at TLSSocket.emit (events.js:228:7)

    at endReadableNT (_stream_readable.js:1185:12)

    at processTicksAndRejections (internal/process/task_queues.js:81:21)

Error #3 --------------------------------------------------------------------------------

Error: socket hang up

    at connResetException (internal/errors.js:570:14)

    at TLSSocket.socketOnEnd (_http_client.js:440:23)

    at TLSSocket.emit (events.js:228:7)

    at endReadableNT (_stream_readable.js:1185:12)

    at processTicksAndRejections (internal/process/task_queues.js:81:21)

Error #4 --------------------------------------------------------------------------------

Error: socket hang up

    at connResetException (internal/errors.js:570:14)

    at TLSSocket.socketOnEnd (_http_client.js:440:23)

    at TLSSocket.emit (events.js:228:7)

    at endReadableNT (_stream_readable.js:1185:12)

    at processTicksAndRejections (internal/process/task_queues.js:81:21)

Error #5 --------------------------------------------------------------------------------

Error: socket hang up

    at connResetException (internal/errors.js:570:14)

    at TLSSocket.socketOnEnd (_http_client.js:440:23)

    at TLSSocket.emit (events.js:228:7)

    at endReadableNT (_stream_readable.js:1185:12)

    at processTicksAndRejections (internal/process/task_queues.js:81:21)  stackTrace=

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error: Cannot cleanup: 

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error #1 --------------------------------------------------------------------------------

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error: socket hang up

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at connResetException (internal/errors.js:570:14)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.socketOnEnd (_http_client.js:440:23)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.emit (events.js:228:7)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at endReadableNT (_stream_readable.js:1185:12)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at processTicksAndRejections (internal/process/task_queues.js:81:21)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error #2 --------------------------------------------------------------------------------

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error: socket hang up

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at connResetException (internal/errors.js:570:14)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.socketOnEnd (_http_client.js:440:23)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.emit (events.js:228:7)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at endReadableNT (_stream_readable.js:1185:12)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at processTicksAndRejections (internal/process/task_queues.js:81:21)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error #3 --------------------------------------------------------------------------------

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error: socket hang up

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at connResetException (internal/errors.js:570:14)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.socketOnEnd (_http_client.js:440:23)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.emit (events.js:228:7)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at endReadableNT (_stream_readable.js:1185:12)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at processTicksAndRejections (internal/process/task_queues.js:81:21)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error #4 --------------------------------------------------------------------------------

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error: socket hang up

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at connResetException (internal/errors.js:570:14)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.socketOnEnd (_http_client.js:440:23)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.emit (events.js:228:7)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at endReadableNT (_stream_readable.js:1185:12)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at processTicksAndRejections (internal/process/task_queues.js:81:21)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error #5 --------------------------------------------------------------------------------

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Error: socket hang up

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at connResetException (internal/errors.js:570:14)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.socketOnEnd (_http_client.js:440:23)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at TLSSocket.emit (events.js:228:7)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at endReadableNT (_stream_readable.js:1185:12)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at processTicksAndRejections (internal/process/task_queues.js:81:21)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at throwError (/home/travis/build/hovancik/stretchly/node_modules/builder-util/src/asyncTaskManager.ts:90:11)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at checkErrors (/home/travis/build/hovancik/stretchly/node_modules/builder-util/src/asyncTaskManager.ts:53:9)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at AsyncTaskManager.awaitTasks (/home/travis/build/hovancik/stretchly/node_modules/builder-util/src/asyncTaskManager.ts:58:5)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at PublishManager.awaitTasks (/home/travis/build/hovancik/stretchly/node_modules/app-builder-lib/src/publish/PublishManager.ts:210:28)

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at /home/travis/build/hovancik/stretchly/node_modules/app-builder-lib/src/index.ts:94:32

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 at executeFinally (/home/travis/build/hovancik/stretchly/node_modules/builder-util/src/promise.ts:25:9)
```

**Comment by b-zurg** [comment]
Does it happen repeatedly? Is it a one off thing?

**Comment by hovancik** [comment]
Well I tried like 10 times now and it seems always the same.

**Comment by hovancik** [comment]
Keep this opened.

**Comment by marc2332** [comment]
Hello 🖖🏻 

I am also facing this issue since some weeks ago.
I now always to re-run the Github Actions until it works.

https://github.com/Graviton-Code-Editor/Graviton-App/runs/1140459243?check_suite_focus=true

Notes:
* It fails randomly
* Only fails in Windows and Ubuntu

**Comment by tassoevan** [comment]
Publishing is almost unusable. It is falling for me at a ratio of around one successful upload to five failed ones.

**Comment by b-zurg** [comment]
I have experienced socket hang ups in different contexts when there is a long running upload. It depends on the specific connection being made and if the connection is not kept alive the upload target thinks it's a dead connection and kills it. 

That's what it seems like is happening here. The comments indicate that it does go through sometimes. I would bet that those times the upload happened just a bit faster. 

I'm also going to guess that is it's under 2 minutes it's fine as the default connection time out for node JS is 2 minutes. 

Could anyone verify these guesses?

**Comment by tassoevan** [comment]
If I'm reading [this](https://github.com/electron-userland/electron-builder/blob/master/packages/electron-publish/src/gitHubPublisher.ts#L163-L205) right, there is no option to set a preferred timeout and it may fail four times before rejecting the upload.

**Comment by tassoevan** [comment]
I've noticed something else: *it doesn't fail at the times my draft release is empty*, i.e., **there was no need to overwrite files**. Looking at all the examples you folks posted here, they all are trying to overwrite files. Is it possible that a change in GitHub API is the root cause?

**Comment by marc2332** [comment]
> 
> 
> I've noticed something else: _it doesn't fail at the times my draft release is empty_, i.e., **when there is no need to overwrite files**. Looking at all the examples you folks posted here, they all are trying to overwrite files. Is it possible that a change in GitHub API is the root cause?

You are absolutely right, it has only been randomly failing when I had a draft matching the same version. Now I have increased the version, it does no longer match the draft's version so it now doesn't randomly fail anymore

**Comment by fspoettel** [comment]
Still relevant

**Comment by 9oelM** [comment]
![Screen Shot 2020-12-30 at 10 46 26 AM](https://user-images.githubusercontent.com/22465806/103324865-6aa4e380-4a8c-11eb-8ea5-d23bbbb9df83.png)
yeap. still.

**Comment by gbmhunter** [comment]
Can also confirm that this is still an issue, and yes, only occurs if the version tag already exists (i.e. it is overwriting a file).

![image](https://user-images.githubusercontent.com/2396869/103452912-6fda8a80-4d39-11eb-9d18-c303fdf19799.png)

**Comment by flesicek** [comment]
same issue here, we are experiencing this error on-off for months :-(

**Comment by algernon** [comment]
Having the same issue: publishing a ~60-80mb package fails on Windows and Linux fairly often, and I have to restart the action. I seem to remember seeing it fail on macOS too, but I restarted the GitHub action, and no longer have the logs to verify. I can verify however what has been observed before: it only fails if there's assets to overwrite. Haven't seen it fail when publishing new assets.

**Comment by flesicek** [comment]
Adding log file failed upload to github from Azure Pipelines

```
==============================================================================
Task         : Bash
Description  : Run a Bash script on macOS, Linux, or Windows
Version      : 3.179.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
==============================================================================
Generating script.
Script contents:
yarn dist
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /Users/runner/work/_temp/4d3341b5-17ab-4376-b393-6d6bde38ab4f.sh
yarn run v1.22.5
$ electron-builder
  • electron-builder  version=22.9.1 os=18.7.0
  • artifacts will be published if draft release exists  reason=CI detected
  • loaded configuration  file=package.json ("build" field)
  • rebuilding native dependencies  dependencies=@tensorflow/tfjs-node@1.5.1, sqlite3@5.0.0 platform=darwin arch=x64
  • rebuilding native dependency  name=sqlite3 version=5.0.0
  • rebuilding native dependency  name=@tensorflow/tfjs-node version=1.5.1
  • packaging       platform=darwin arch=x64 electron=8.5.2 appOutDir=artifacts/mac
  • downloading     url=https://github.com/electron/electron/releases/download/v8.5.2/electron-v8.5.2-darwin-x64.zip size=66 MB parts=6
  • downloaded      url=https://github.com/electron/electron/releases/download/v8.5.2/electron-v8.5.2-darwin-x64.zip duration=1.161s
  • signing         file=artifacts/mac/XXX
identityHash=XXX provisioningProfile=none
Notorization skipped in the CI
  • building        target=macOS zip arch=x64 file=artifacts/xxx.zip
  • building        target=DMG arch=x64 file=artifacts/xxx.dmg
  • building embedded block map  file=artifacts/xxx.zip
  • publishing      publisher=Github (owner: iERP-ai, project: studio, version: 0.9.4-dev)
  • uploading       file=xxx.zip provider=GitHub
  • building block map  blockMapFile=artifacts/xxx.dmg.blockmap
  • uploading       file=xxx.dmg.blockmap provider=GitHub
  • uploading       file=xxx.dmg provider=GitHub
  ⨯ write EPIPE     stackTrace=
                      Error: write EPIPE
                          at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:92:16)
                          at handleWriteReq (internal/stream_base_commons.js:51:26)
                          at writeGeneric (internal/stream_base_commons.js:143:15)
                          at TLSSocket.Socket._writeGeneric (net.js:784:11)
                          at TLSSocket.Socket._write (net.js:796:8)
                          at doWrite (_stream_writable.js:403:12)
                          at doWrite (_stream_writable.js:403:12)
                          at clearBuffer (_stream_writable.js:542:7)
                          at TLSSocket.Writable.uncork (_stream_writable.js:338:7)
                          at ClientRequest.OutgoingMessage.uncork (_http_outgoing.js:244:17)
                          at connectionCorkNT (_http_outgoing.js:698:8)
                          at processTicksAndRejections (internal/process/task_queues.js:84:21)
  • overwrite published file  file=xxx.dmg.blockmap reason=already exists on GitHub
  • overwrite published file  file=xxx.dmg reason=already exists on GitHub
error Command failed with exit code 1.

```

**Comment by GiancarlosIO** [comment]
Same here 😢

**Comment by scottrobertson** [comment]
Also facing this

**Comment by scottrobertson** [comment]
> Can also confirm that this is still an issue, and yes, only occurs if the version tag already exists (i.e. it is overwriting a file).

Could the script perhaps try and delete the file first, then upload?

**Comment by 9oelM** [comment]
> > Can also confirm that this is still an issue, and yes, only occurs if the version tag already exists (i.e. it is overwriting a file).
> 
> Could the script perhaps try and delete the file first, then upload?

https://github.com/electron-userland/electron-builder/blob/0ba839b72b6a13178ad5bc4b03a47f9aca980650/packages/electron-publish/src/gitHubPublisher.ts#L186-L188

I'm afraid if it is already doing it. If I have some time later i might dig into this. I might be wrong, but i'm suspecting it might be some changes in Github API that breaks the if statement or something 
https://github.com/electron-userland/electron-builder/blob/0ba839b72b6a13178ad5bc4b03a47f9aca980650/packages/electron-publish/src/gitHubPublisher.ts#L186

**Comment by scottrobertson** [comment]
> > > Can also confirm that this is still an issue, and yes, only occurs if the version tag already exists (i.e. it is overwriting a file).
> > 
> > 
> > Could the script perhaps try and delete the file first, then upload?
> 
> https://github.com/electron-userland/electron-builder/blob/0ba839b72b6a13178ad5bc4b03a47f9aca980650/packages/electron-publish/src/gitHubPublisher.ts#L186-L188
> 
> I'm afraid if it is already doing it. I'm not even sure if this is the right place to look at because i have not contributed to this repo at all, but If I have some time later i might dig into this. I might be wrong, but i'm suspecting it might be some changes in Github API that breaks the if statement or something
> https://github.com/electron-userland/electron-builder/blob/0ba839b72b6a13178ad5bc4b03a47f9aca980650/packages/electron-publish/src/gitHubPublisher.ts#L186

I think you are right. I can take a look too, just trying to work out how to run this project from a fork right now.

**Comment by scottrobertson** [comment]
> > > Can also confirm that this is still an issue, and yes, only occurs if the version tag already exists (i.e. it is overwriting a file).
> > 
> > 
> > Could the script perhaps try and delete the file first, then upload?
> 
> https://github.com/electron-userland/electron-builder/blob/0ba839b72b6a13178ad5bc4b03a47f9aca980650/packages/electron-publish/src/gitHubPublisher.ts#L186-L188
> 
> I'm afraid if it is already doing it. I'm not even sure if this is the right place to look at because i have not contributed to this repo at all, but If I have some time later i might dig into this. I might be wrong, but i'm suspecting it might be some changes in Github API that breaks the if statement or something
> https://github.com/electron-userland/electron-builder/blob/0ba839b72b6a13178ad5bc4b03a47f9aca980650/packages/electron-publish/src/gitHubPublisher.ts#L186

Na, it's getting into that loop.

```
  • overwrite published file  file=Balance-Bar-Setup-0.5.0.exe.blockmap reason=already exists on GitHub
  ⨯ socket hang up  stackTrace=
                      Error: socket hang up
                          at connResetException (internal/errors.js:607:14)
                          at TLSSocket.socketOnEnd (_http_client.js:493:23)
                          at TLSSocket.emit (events.js:327:22)
                          at endReadableNT (internal/streams/readable.js:1327:12)
                          at processTicksAndRejections (internal/process/task_queues.js:80:21)
```

**Comment by 9oelM** [comment]
Well yeah. perhaps related: https://github.com/request/request/issues/2047

Regardless of whether we use `request` i think it just might be a general issue across node.js applications. Based on the thread, we may want to try fiddling around with request headers.

**Comment by 9oelM** [comment]
oops. link to the request issue was mentioned at the top. sorry

**Comment by TheOtterlord** [comment]
Getting this near 100% of the time on a new project I contribute towards.

Is there any progress on a custom option or a way to prevent the timeout?
Or perhaps a fork of the project that has solved this?

**Comment by b-zurg** [comment]
> Getting this near 100% of the time on a new project I contribute towards.
> 
> Is there any progress on a custom option or a way to prevent the timeout?
> Or perhaps a fork of the project that has solved this?

Try building / publishing in GitHub actions. I've never had an issue there. I really think it's something dependent on the speed of the internet connection

**Comment by mikekreeki** [comment]
Using CI like Github Actions does not resolve this issue. I'm as well getting the same error all the time in my project which building and releasing from GitHub Actions.

**Comment by TheOtterlord** [comment]
@b-zurg it happens in gh actions. The only fix we've found do far is to rerun the action until all files are built.

**Comment by Kellojo** [comment]
I also have this issue quite regularly, mainly, when building for a release that already has some release assets attached (i.e. overwrites the existing ones). A workaround I have used for now is to delete the release assets during the build. This is by no means a proper solution to the problem but a reasonably nice working workaround 😄 

To make this a bit easier I created a [custom GitHub action](https://github.com/Kellojo/delete-release-assets-action) for this. It ready the current version from the package.json and deletes any release assets attached to it. As a small failsafe, it by default only does this for draft releases. All you need to do is to add this to your build action **before electron-builder** runs and **after your repo has been cloned** :

      - name: Delete current release assets
        uses: Kellojo/delete-release-assets-action@v1.8
        with:
          delete-only-drafts: true
          github-access-token: ${{ secrets.github_token }}

**Comment by mmaietta** [comment]
Have you tried that timeout change yet, by chance? 
Here's a quick patch to apply that change for v22.10.5 in case anyone else interested.
If it works, then I'm happy to incorporate the change
```
diff --git a/node_modules/electron-publish/out/gitHubPublisher.js b/node_modules/electron-publish/out/gitHubPublisher.js
index 75e5ca2..fa5ffa8 100644
--- a/node_modules/electron-publish/out/gitHubPublisher.js
+++ b/node_modules/electron-publish/out/gitHubPublisher.js
@@ -247,6 +247,7 @@ class GitHubPublisher extends _publisher().HttpPublisher {
       hostname: parsedUrl.hostname,
       path: parsedUrl.path,
       method: "POST",
+      timeout: 120 * 1000,
       headers: {
         accept: "application/vnd.github.v3+json",
         "Content-Type": _mime().default.getType(fileName) || "application/octet-stream",
```

**Comment by Kellojo** [comment]
I have just now created a pr for this.
Unfortunately I am not sure on how to test it. 
Installing the changed version by adding electron-builder via a GitHub dependency gives me (`npm i git+https://github.com/Kellojo/electron-builder.git`):
```
npm i git+https://github.com/Kellojo/electron-builder.git
npm ERR! premature close

npm ERR! A complete log of this run can be found in:
```

Does anyone have an idea how we could verify this indeed solves the issue?

**Comment by mmaietta** [comment]
@Kellojo you can use https://github.com/wclr/yalc to set up the dependency store locally with your compiled changes. That's how my dev env is set up.

**Comment by mmaietta** [comment]
Bump. Has anyone else tried this patch? https://github.com/electron-userland/electron-builder/issues/4940#issuecomment-813856098

**Comment by hovancik** [comment]
Not as I am building on CI

**Comment by gniezen** [comment]
@mmaietta I tried the patch, it did not work for me. Still getting the "socket hang up" error every time on builds with existing assets.

**Comment by mmaietta** [comment]
@hovancik you can commit the patch as a file to your repo and use https://www.npmjs.com/package/patch-package to have it applied in the `postinstall` stage, this allows patches to persist across installations and devices :)
I use it often when waiting on a PR to be accepted and/or released upstream

That's unfortunate that it's not working for you @gniezen. Would you be willing to try a much larger timeout value than the patch has?  How big are the assets you're uploading?

**Comment by gniezen** [comment]
@mmaietta I used `patch-package`, so I can second that it works really well to apply patches for `electron-builder` :smile: 

I did try larger timeout values, up to 4 minutes, but that didn't help. The asset sizes are between 69MB (.zip) and 146MB (.exe). Here's a gist when I set the timeout to 240s (you'll notice I also included a `Connection: keep-alive`, but that didn't help either): https://gist.github.com/gniezen/9d584ab27845adaf98746cdaf498426c

**Comment by Kellojo** [comment]
I am wondering what the issue could be, if it's not the timeout.
Magically my builds now work again without any issues...

**Comment by mmaietta** [comment]
Could anyone try tracing the network request and see if it provides any more info than electron-builder's confusing logging strategy?

**Comment by gniezen** [comment]
@mmaietta Sure, let me know what you'd like me to do. So far I did a `tcpdump` of the traffic and I can see the GitHub server closing the connection while we're still sending data, resulting in a stream of `[RST, ACK]` packets, where the server acknowledges the packet but that the connection is closed. This appears to be happening less than a second after the connection is opened.

**Comment by vraravam** [comment]
I'm also seeing this error. Unfortunately, we are not using GH Actions (we are on travisCI). But the same "socket connection" issue

**Comment by tangb** [comment]
I also encountered this issue on both CircleCI and Bitrise.

**Comment by bootsie123** [comment]
Also encountering this on Travis CLI when updating assets on a pre-existing tag

**Comment by Kellojo** [comment]
Maybe we could try to retry the upload if it fails due to this error. While it might not resolve the core issue it could make it less prominent. Especially since it always seems to hang up randomly without an aparent reason.

**Comment by vraravam** [comment]
We have been using `travis-ci`, which has an inbuilt `travis_retry` function. Even with this, the socket hangup issue occurs - and all it has effectively done is increase our build times by 3x - since each retry *still fails at the same spot* for one or the other artifact being uploaded.

**Comment by tassoevan** [comment]
I'm trying to debug the issue at version 22.11.7 using the strategy of patching the `electron-publish` package. I'm running `electron-builder` using GitHub Actions. This is the error thrown:

```
HttpError: 502 Bad Gateway
{
  "message": "Error creating policy",
  "request_id": "0403:2E2B:1F4F46:2A1335:60C4F970"
}
Headers: {
  "cache-control": "no-cache",
  "content-length": "83",
  "content-security-policy": "default-src 'none'",
  "content-type": "application/json; charset=utf-8",
  "strict-transport-security": "max-age=31557600",
  "x-content-type-options": "nosniff",
  "x-frame-options": "deny",
  "x-xss-protection": "1; mode=block",
  "date": "Sat, 12 Jun 2021 18:14:09 GMT",
  "x-github-request-id": "0403:2E2B:1F4F46:2A1335:60C4F970",
  "connection": "close"
}
    at createHttpError (D:\a\Rocket.Chat.Electron\Rocket.Chat.Electron\node_modules\builder-util-runtime\src\httpExecutor.ts:30:10)
    at IncomingMessage.<anonymous> (D:\a\Rocket.Chat.Electron\Rocket.Chat.Electron\node_modules\builder-util-runtime\src\httpExecutor.ts:179:18)
    at IncomingMessage.emit (events.js:388:22)
    at endReadableNT (internal/streams/readable.js:1336:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  statusCode: 502,
  description: {
    message: 'Error creating policy',
    request_id: '0403:2E2B:1F4F46:2A1335:60C4F970'
  },
  code: 'HTTP_ERROR_502'
}
```

**Comment by vraravam** [comment]
Since I posted my earlier messages, we moved Ferdi from `travis-ci` and `appveyor` to GH Actions. We are still facing the same issue as above - with electron-builder on 22.11.7. (I haven't patched anything yet)

**Comment by tassoevan** [comment]
The best I can infer from looking the code: doing parallel uploads for GitHub is impossible, even inside the environment of GitHub Actions. We're hitting at some rate limit due to how `electron-builder` calls tasks from `electron-publish`. Unfortunately I don't have the time and the knowledge to make the uploads run as a serial chain.

**Comment by vraravam** [comment]
I can confirm [this comment](https://github.com/electron-userland/electron-builder/issues/4940#issuecomment-698738497) - I have seen it fail consistently when trying to overwrite pre-existing artifacts that are in `draft` mode. And the instances where this "socket hangup" issue has occurred when no draft version is present - this is at *zero* for our codebase.

**Comment by jqueryisamonad** [comment]
Hello friends!

I see that you passed a timeout in the request, but it looks like it was overwritten by the `addTimeOutHandler` method in `httpExecutor`.
https://github.com/electron-userland/electron-builder/blob/c9b78f958daf778747843dd2d671628f9cbb8ef6/packages/builder-util-runtime/src/httpExecutor.ts#L301

So I changed the timeout value in `addTimeOutHandler` method, applied the patch with `patch-package` and ran the builder about 10 times on GitHub Actions and never got the `socket hang up` error. Maybe I just got lucky, but it would be nice if other people tested it.

So, here's my `builder-util-runtime+8.7.7.patch` for [patch-package](https://github.com/ds300/patch-package#readme):
```diff
diff --git a/node_modules/builder-util-runtime/out/httpExecutor.js b/node_modules/builder-util-runtime/out/httpExecutor.js
index f283a46..da45fd1 100644
--- a/node_modules/builder-util-runtime/out/httpExecutor.js
+++ b/node_modules/builder-util-runtime/out/httpExecutor.js
@@ -245,10 +245,13 @@ Please double check that your authentication token is correct. Due to security r
     }
     addTimeOutHandler(request, callback) {
         request.on("socket", (socket) => {
-            socket.setTimeout(60 * 1000, () => {
+            socket.setTimeout(3 * 60 * 1000, () => {
                 request.abort();
                 callback(new Error("Request timed out"));
             });
         });
     }
     static prepareRedirectUrlOptions(redirectUrl, options) {
```

🤞

**Comment by jmrossy** [comment]
+1 for this issue.

I'm using Github actions to build + release for Windows, Linux, and Mac. The Linux (Ubuntu) build almost never fails. The Windows and Mac builds fail most of the time. I'm currently on my longest running fail streak of 30+ tries but eventually the jobs do work.

Here's the relevant error message:
```
  • downloading     url=https://github.com/electron/electron/releases/download/v13.1.1/electron-v13.1.1-darwin-x64.zip size=79 MB parts=6
  • downloaded      url=https://github.com/electron/electron/releases/download/v13.1.1/electron-v13.1.1-darwin-x64.zip duration=1.367s
  • downloading     url=*** size=3.2 kB parts=1
  ⨯ part download request failed with status code 403
  ```

**Comment by RobertLowe** [comment]
Between this and https://github.com/electron-userland/electron-builder/issues/4299 releasing on macos is really in a sad state of affairs

I can also confirm this only appears to be happening with existing draft releases.

**Comment by mandrasch** [comment]
I had multiple socket hangups as well (https://github.com/mandrasch/not-a-real-screenreader/blob/main/.github/workflows/build.yml) while overwriting existings draft files. I manually deleted the asset files from release draft, rerunning the jobs now worked twice successfully. <img width="708" alt="Edit_release_·_mandrasch_not-a-real-screenreader" src="https://user-images.githubusercontent.com/777278/125193935-50c03680-e24f-11eb-99e5-c976eafda0bd.png">

**Comment by vraravam** [comment]
Yes @mandrasch - that is a manual workaround (which we have exercised a couple of times) - and one that is not sustainable. Our usecase is that the CI build kicks off every 24 hours, we would not want to be obligated to manually delete each and every build artefact like this.

**Comment by RobertLowe** [comment]
> Yes @mandrasch - that is a manual workaround (which we have exercised a couple of times) - and one that is not sustainable. Our usecase is that the CI build kicks off every 24 hours, we would not want to be obligated to manually delete each and every build artefact like this.

I suggest automating with the `github-cli` https://cli.github.com/manual/gh_release_delete to do clean up or https://cli.github.com/manual/gh_release_upload overwrite assets

**Comment by vraravam** [comment]
Thanks for the tips @RobertLowe - but, shouldn't this be fixed within `electron-builder` itself? Why do we need to propagate a band-aid for a bug in the original system?

**Comment by RobertLowe** [comment]
@vraravam ideally yes, but at least you have a workable solution until such time, maybe try making a pull request?

**Comment by mmaietta** [comment]
Looks like the upload+overwrite feature was implemented. Maybe github API responses changed such that this executing?

**Comment by sheck** [comment]
I'm still seeing this on an electron project I work on. When we run parallel windows and Mac builds on GH actions that publishes to GH releases, it seems to fail about 60% of the time with this error.

We've since switched to serialized builds and haven't encountered the issue anymore, but it would be nice to have parallel builds again.

**Comment by phanirithvij2000** [comment]
I can reproduce it still, for parallel builds

**Comment by VictorioBerra** [comment]
Happening to me too, trying to deploy on Windows and Ubuntu.

**Comment by danger89** [comment]
still have this happening multiple times today.

**Comment by RobertLowe** [comment]
> [](https://github.com/lbry-foss/lbry-desktop/runs/4692765872?check_suite_focus=true#step:6:207)
> 
> Is it a time-out we are hitting or?
> 
> * Did you check for setting `keepAlive: true` in the http agent?

If I recall correctly, this is from the github API, not the actual http client; Further, I had difficulties around this because of draft mode. Even after changing the release from draft mode, the issue remained. There is some lingering funk from draft mode, GH doesnt clean-up well there?

**Comment by danger89** [comment]
If there is some error coming from the github api. Electron builder should handle those errors as well. Avoiding socket closed issue. Maybe we then also see what the root cause may be.

**Comment by baparham** [comment]
I have these observations from trying to debug this locally:

1. we never enter the `if` block where `overwriteArtifact` is called

https://github.com/electron-userland/electron-builder/blob/00e49e476ab103028d3e8d2543e34c041bc72878/packages/electron-publish/src/gitHubPublisher.ts#L200-L202

2. the description we get back from github is not json, but plain text, so the `e.description.errors[0].code` in the if statement is undefined since `e.description` has not been parsed into an object.
3. I do not believe this is the entirety of this problem, since I continue to _also_ hit 502 bad gateway or other econnreset errors sometimes alongside, however this may be the root cause of the "can't overwrite an already created draft release asset" problem

I'll continue experimenting and debugging, and try and come up with a PR to fix at least this portion of these errors

the description we get back is like:
```
 description: 'method: POST url: https://uploads.github.com/repos/......./releases/......./assets?name=......exe\n' +
    '\n' +
    '          Data:\n' +
    '          {"message":"Validation Failed","request_id":".......","documentation_url":"https://docs.github.com/rest","errors":[{"resource":"ReleaseAsset","code":"already_exists","field":"name"}]}\n' +
    '          ',
```

**Comment by gniezen** [comment]
I've tried electron-builder v23.0.2, which has the fix in #6563 , but unfortunately this leads to a never-ending loop on CircleCI builds (but works fine on Appveyor):

```
overwrite published file  file=Tidepool Uploader-2.43.0-****************-update.1-mac.zip.blockmap reason=already exists on GitHub
  **************** Request: {
  ****************   "protocol": "https:",
  ****************   "hostname": "api.github.com",
  ****************   "port": null,
  ****************   "path": "/repos/tidepool-org/uploader/releases/61378130/assets",
  ****************   "headers": {
  ****************     "accept": "application/vnd.github.v3+json",
  ****************     "authorization": "<stripped sensitive data>",
  ****************     "User-Agent": "****************",
  ****************     "Cache-Control": "no-cache"
  ****************   }
  **************** } +1ms
  **************** Response: 200 OK, request options: {
  ****************   "protocol": "https:",
  ****************   "hostname": "api.github.com",
  ****************   "port": null,
  ****************   "path": "/repos/tidepool-org/uploader/releases/61378130/assets",
  ****************   "headers": {
  ****************     "accept": "application/vnd.github.v3+json",
  ****************     "authorization": "<stripped sensitive data>",
  ****************     "User-Agent": "****************",
  ****************     "Cache-Control": "no-cache"
  ****************   }
  **************** } +155ms
  • trying to upload again  file=Tidepool Uploader-2.43.0-****************-update.1-mac.zip.blockmap reason=not found on GitHub
  **************** Request: {
  ****************   "protocol": "https:",
  ****************   "hostname": "uploads.github.com",
  ****************   "path": "/repos/tidepool-org/uploader/releases/61378130/assets?name=Tidepool%20Uploader-2.43.0-****************-update.1-mac.zip.blockmap",
  ****************   "method": "POST",
  ****************   "headers": {
  ****************     "accept": "application/vnd.github.v3+json",
  ****************     "Content-Type": "application/octet-stream",
  ****************     "Content-Length": 75743,
  ****************     "authorization": "<stripped sensitive data>",
  ****************     "User-Agent": "****************",
  ****************     "Cache-Control": "no-cache"
  ****************   }
  **************** } +25ms
  **************** Response: 422 Unprocessable Entity, request options: {
  ****************   "protocol": "https:",
  ****************   "hostname": "uploads.github.com",
  ****************   "path": "/repos/tidepool-org/uploader/releases/61378130/assets?name=Tidepool%20Uploader-2.43.0-****************-update.1-mac.zip.blockmap",
  ****************   "method": "POST",
  ****************   "headers": {
  ****************     "accept": "application/vnd.github.v3+json",
  ****************     "Content-Type": "application/octet-stream",
  ****************     "Content-Length": 75743,
  ****************     "authorization": "<stripped sensitive data>",
  ****************     "User-Agent": "****************",
  ****************     "Cache-Control": "no-cache"
  ****************   }
  **************** } +1s
  • overwrite published file  file=Tidepool Uploader-2.43.0-****************-update.1-mac.zip.blockmap reason=already exists on GitHub
```

**Comment by chrisdevereux** [comment]
Upgrading from 22.14 -> 23.0 fixed this for me. Thanks!

**Comment by gniezen** [comment]
@mmaietta I think you may have closed this one in error. The fix works on Windows/Appveyor builds, but (at least for me) led to a never-ending loop on MacOS/CircleCI builds.

**Comment by MrGoro** [comment]
I can confirm that for me the error still exists with Version 23.0.3. It randomly fails uploading to existing assets. After some reruns it works.

**Comment by Deathspike** [comment]
This definitely still occurs in 23.0.3.

**Comment by paustint** [comment]
😭  just found this thread - still happening `23.0.9`

**Comment by mmaietta** [comment]
Reopening. I'm going to need someone else to investigate this though as I have no idea where to debug this.

**Comment by ocavue** [comment]
GitHub's official  `actions/upload-artifact` seems to resolve this issue by adding retries:

https://github.com/actions/upload-artifact/issues/116#issuecomment-754141048

> Should be good now. Retries with exponential back-off have been added to all HTTP calls so any socket disconnects, timeouts, 503s or `ECONNRESET` errors should gracefully now retry and the artifact upload should go through (previously only a subset of HTTP calls during upload were retried). Recent updates should significantly improve reliability.

**Comment by shensven** [comment]
same problem at `23.3.1`

**Comment by koningcool** [comment]
Any updates on this?

**Comment by mmaietta** [comment]
Happy to review a PR contribution for this issue as I'm not able to devote any of my time to this ticket.

**Comment by koningcool** [comment]
Can you explain why? And unfortunately I am not able to fix this, I simply don’t know how.

**Comment by Deathspike** [comment]
Not sure if this helps for your immediate problem @koningcool, but I have a workaround in place that simply deletes the release before running `electron-builder`. It's relatively straightforward to implement if you're using *Github* workflows. I also took a stab at this way back when, but I also have no idea how to reproduce the issue in a reliable way in order to fix it. For my workaround, see:

https://github.com/Deathspike/ulmas/blob/78b42f70e0c940e9731cc66fb87bdb09f687f160/.github/workflows/electron-snapshot.yml
https://github.com/Deathspike/ulmas/blob/78b42f70e0c940e9731cc66fb87bdb09f687f160/.github/scripts/electron-snapshot-prune.js
